### PR TITLE
CircleCI config: add LocalCI support

### PR DESCRIPTION
**Pull request by hewsut**
LocalCI is a localization continuous integration package that coordinates reporting between a CI system (Circle in our case) and Github. This PR adds pot diffing (using gettext CLI tools) to generate the data necessary to feed into LocalCI and a webhook to trigger the integration.

See: https://github.com/Automattic/gp-localci

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -24,4 +24,15 @@ test:
   post:
     - mkdir -p $CIRCLE_TEST_REPORTS/junit/ && find . -type f -regex  "./test-results.*\.xml" -exec cp {} $CIRCLE_TEST_REPORTS/junit/ \;:
         parallel: true
-    - if [[ $CIRCLE_BRANCH == "master" ]]; then make translate; mkdir $CIRCLE_ARTIFACTS/translate; cp calypso-strings.pot $CIRCLE_ARTIFACTS/translate/calypso-strings.pot; fi;
+    - >
+        make translate;
+        mkdir $CIRCLE_ARTIFACTS/translate;
```
Comment (deBhal) [inline circle.yml]: We should add a -p

**Comment by deBhal** [comment]
LGTM!

**Comment by hewsut** [comment]
Got the LGTM from @deBhal on Slack. Merging.

**Comment by gwwar** [comment]
@hewsut do we need to run `make translate` for all branches? If we're not using the branch specific artifacts, let's limit this to only master since we'll be increasing build times.

**Comment by deBhal** [comment]
@gwwar: We're building tools to improve our handling of strings before merge, so we definitely do want to run `make translate` in non-master branches.

We don't want to blow out CircleCI, though, and we've got a range of options to reduce the impact if it becomes a problem.

@hewsut: Actually, having a look at an example build timing, (e.g. [#](https://circleci.com/gh/Automattic/wp-calypso/31864#build-timing/containers/0/actions/8)), I think I was wrong about running on a secondary container. We'd need to make sure we do add in logic to check if it's a parallel build, but as long as we do that, I think we could run make translate alongside/after the build-server step. I don't think we need the tests or the build for the translatable strings.

automattic_wp-calypso__31864_-_circleci

**Comment by hewsut** [comment]
@gwwar Yes. We have to do it to discover new strings being added. The plan is to report back to GH UI translated/untranslated information. FWIW, it seems to add a mere ~8 sec to build times if I tracked the change correctly.

**Comment by gwwar** [comment]
> We're building tools to improve our handling of strings before merge

Gotcha, thanks for the clarification.

>  FWIW, it seems to add a mere ~8 sec to build times

It is pretty short, but it can add up if we have a number of builds in the queue. Just wanted to double check and see if we needed it for all branches. Previously the pot file was only generated during deploy in a build script.

**Comment by hewsut** [comment]
Yep.

I think we're going to have to tweak the config again (won't add time, just some options to `msgcat`) so maybe I will look into running it on another node during server build like Jules suggested. That should make its added time zero.

**Pull request merged**